contracts/libs/Ibc.sol

```diff
@@ -305,13 +305,16 @@ library Ibc {
         returns (address emittingContract, bytes[] memory topics, bytes memory unindexedData)
     {
         // The first byte is a RLP encoded receipt type so slice it off.
-        RLPReader.RLPItem[] memory receipt;
-        if (receiptRLP.length % 2 == 0) {
-            // If even, we don't clip bytes
-            receipt = RLPReader.readList(receiptRLP);
-        } else {
-            receipt = RLPReader.readList(Bytes.slice(receiptRLP, 1, receiptRLP.length - 1));
+        uint8 typeByte;
+        assembly {
+            typeByte := byte(0, mload(add(receiptRLP, 32)))
+        }
+        if (typeByte < 0x80) {
+            // Typed receipt: strip the type byte
+            receiptRLP = Bytes.slice(receiptRLP, 1, receiptRLP.length - 1);
```

Comment:
```suggestion
            receipt =  RLPReader.readList(Bytes.slice(receiptRLP, 1, receiptRLP.length - 1));
            }
            else{
            receipt =  RLPReader.readList(receiptRLP);
            }
```

Worried a tad about a potential stray byte at the end  caused by overwriting the receiptRLP itself which technically isn't being cleared. Shouldn't be a problem since we overwrite the length part but just to be safe as we're doing some assembly in this one